returnn/datasets/concat_files.py

```diff
@@ -159,7 +199,9 @@ def _lazy_init_num_outputs(self):
     def _lazy_init_file_sizes(self):
         if self._file_sizes:
             return
-        self._file_sizes = {fn: os.path.getsize(fn) for fn in self.files}
+
+        import tree
+        self._file_sizes = {":".join(tree.flatten(t)): sum((os.path.getsize(fn) for fn in tree.flatten(t)), 0) for t in self.files}
```

Comment:
I think we should not use such ad hoc key format.
```suggestion
        self._file_sizes = [sum((os.path.getsize(fn) for fn in tree.flatten(t)), 0) for t in self.files]
```